src/tla2528.cpp

```diff
@@ -108,72 +114,75 @@ void tla2528::set_pin_mode(pin_mode p_mode, hal::byte p_channel)
+  // will reference the desired state cache.
   m_gpo_value = p_values;
   std::array<hal::byte, 3> cmd_buffer = { op_codes::single_register_write,
                                           register_addresses::gpo_value,
                                           m_gpo_value };
   hal::write(m_i2c_bus, m_i2c_address, cmd_buffer);
 }
 
-void tla2528::set_digital_out(hal::byte p_channel, bool p_high)
+void tla2528::set_output_pin(hal::byte p_channel, bool p_high)
 {
   throw_if_invalid_channel(p_channel);
   if (p_high) {
     hal::bit_modify(m_gpo_value).set(hal::bit_mask::from(p_channel));
   } else {
     hal::bit_modify(m_gpo_value).clear(hal::bit_mask::from(p_channel));
   }
-  set_digital_bus_out(m_gpo_value);
+  set_output_bus(m_gpo_value);
 }
 
-bool tla2528::get_digital_out(hal::byte p_channel)
+bool tla2528::get_output_pin_state(hal::byte p_channel)
 {
   throw_if_invalid_channel(p_channel);
   return hal::bit_extract(hal::bit_mask::from(p_channel),
-                          get_digital_bus_out());
+                          get_output_bus_state());
 }
-hal::byte tla2528::get_digital_bus_out()
+hal::byte tla2528::get_output_bus_state()
 {
   std::array<hal::byte, 1> data_buffer;
-  std::array<hal::byte, 2> cmd_buffer = {
+  constexpr std::array<hal::byte, 2> cmd_buffer = {
     op_codes::single_register_read,
     register_addresses::gpo_value,
   };
   hal::write_then_read(m_i2c_bus, m_i2c_address, cmd_buffer, data_buffer);
   return data_buffer[0];
 }
 
-hal::byte tla2528::get_digital_bus_in()
+hal::byte tla2528::get_input_bus()
 {
   std::array<hal::byte, 1> data_buffer;
-  std::array<hal::byte, 3> cmd_buffer = {
+  constexpr std::array<hal::byte, 3> cmd_buffer = {
     op_codes::single_register_read,
     register_addresses::gpi_value,
   };
   hal::write_then_read(m_i2c_bus, m_i2c_address, cmd_buffer, data_buffer);
   return data_buffer[0];
 }
 
-bool tla2528::get_digital_in(hal::byte p_channel)
+bool tla2528::get_input_pin(hal::byte p_channel)
 {
   throw_if_invalid_channel(p_channel);
-  return hal::bit_extract(hal::bit_mask::from(p_channel), get_digital_bus_in());
+  return hal::bit_extract(hal::bit_mask::from(p_channel), get_input_bus());
 }
 
-float tla2528::get_analog_in(hal::byte p_channel)
+float tla2528::get_adc_reading(hal::byte p_channel)
 {
   set_analog_channel(p_channel);
-  // TODO: look into averaging & channel validation
+  // TODO(#8): look into averaging & channel validation
   std::array<hal::byte, 2> data_buffer;
   std::array<hal::byte, 1> cmd_buffer = { op_codes::single_register_read };
   hal::write_then_read(m_i2c_bus, m_i2c_address, cmd_buffer, data_buffer);
 
   // Take 12 bit number stored in first 12 bits of 2 bytes and converting to 16
   // bit num by shifting 4 bit right (See Figure 25 on datasheet)
   uint16_t data = data_buffer[0] << 4 | data_buffer[1] >> 4;
-  return data / 4095.0;
+  return (float)data / (float)4095.0;
```

Comment:
Use C++ style casts (clang-tidy should have caught this). 
```suggestion
  return static_cast<float>(data) / 4095.0f;
```